integration_test/trackedusersreporting/tracked_users_reporting_test.go

```diff
@@ -85,55 +89,112 @@ func TestTrackedUsersReporting(t *testing.T) {
 		return tc.webhook.RequestsCount() == eventsCount
 	}, 1*time.Minute, 5*time.Second, "unexpected number of events received, count of events: %d", tc.webhook.RequestsCount())
 
-	// TODO: once flusher is implemented, add a mock reporting server to check is we are getting correct cardinality of users
-	cardinalityMap := getCardinalityFromDB(t, tc.postgresResource)
-	require.Equal(t, 2, cardinalityMap[workspaceID][sourceID])
+	require.Eventuallyf(t, func() bool {
+		cardinalityMap := getCardinalityFromReportingServer(t, tc.reportingServer.Requests())
+		return cardinalityMap[workspaceID][sourceID].userIDCount == 2 &&
+			cardinalityMap[workspaceID][sourceID].anonIDCount == 2 &&
+			cardinalityMap[workspaceID][sourceID].identifiedUsersCount == 2
+	}, 2*time.Minute, 5*time.Second, "data not reported to reporting service, count of reqs: %d", tc.reportingServer.RequestsCount())
+
 	cancel()
 	require.NoError(t, wg.Wait())
 }
 
-func getCardinalityFromDB(t *testing.T, postgresResource *postgres.Resource) map[string]map[string]int {
+func getCardinalityFromReportingServer(t *testing.T, reqs []*http.Request) map[string]map[string]struct {
```

Comment:
I would use a struct that embeds `*webhookutil.Recorder` and implement this function as a method.

Or could you consider creating a dedicated recorder with specific domain logic? This can simplify the implementation, and allow validation during the HTTP request.